File: backend/routes/parent.routes.js
import { Router } from 'express';
import {
  getAllParents,
  getParentById,
  createParent,
  updateParent,
  deleteParent,
} from '../controllers/parent.controller.js';
import { verifyToken, requireRole } from '../middleware/auth.middleware.js';

const router = Router();

router.use(verifyToken);

router.get('/',    requireRole('admin', 'teacher'), getAllParents);
router.get('/:id', getParentById);
router.post('/',   requireRole('admin'), createParent);
router.put('/:id', requireRole('admin'), updateParent);
router.delete('/:id', requireRole('admin'), deleteParent);

export default router;

File: backend/routes/parent_routes.py
from flask import Blueprint
from controllers.parent_controller import (
    get_all_parents,
    get_parent_by_id,
    create_parent,
    update_parent,
    delete_parent,
)
from middleware.auth_middleware import verify_token, require_role

parent_bp = Blueprint('parents', __name__)

parent_bp.route('/', methods=['GET'])(
    verify_token(require_role('admin', 'teacher')(get_all_parents))
)
parent_bp.route('/<string:parent_id>', methods=['GET'])(
    verify_token(get_parent_by_id)
)
parent_bp.route('/', methods=['POST'])(
    verify_token(require_role('admin')(create_parent))
)
parent_bp.route('/<string:parent_id>', methods=['PUT'])(
    verify_token(require_role('admin')(update_parent))
)
parent_bp.route('/<string:parent_id>', methods=['DELETE'])(
    verify_token(require_role('admin')(delete_parent))
)
